src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppElasticBeanstalk.recipe

```diff
@@ -329,6 +329,126 @@
                     ]
                 }
             ]
+        },
+        {
+            "Id": "ElasticBeanstalkRollingUpdates",
+            "Name": "Rolling Updates",
+            "Description": "When a configuration change requires replacing instances, Elastic Beanstalk can perform the update in batches to avoid downtime while the change is propagated. During a rolling update, capacity is only reduced by the size of a single batch, which you can configure. Elastic Beanstalk takes one batch of instances out of service, terminates them, and then launches a batch with the new configuration. After the new batch starts serving requests, Elastic Beanstalk moves on to the next batch.",
+            "Type": "Object",
+            "AdvancedSetting": true,
+            "Updatable": true,
+            "DependsOn": [
+                {
+                    "Id": "EnvironmentType",
+                    "Value": "LoadBalanced"
+                }
+            ],
+            "ChildOptionSettings": [
+                {
+                    "Id": "RollingUpdatesEnabled",
+                    "Name": "Enable Rolling Updates",
+                    "Description": "Do you want to enable Rolling Updates?",
+                    "Type": "Bool",
+                    "DefaultValue": false,
+                    "AdvancedSetting": true,
+                    "Updatable": true
+                },
+                {
+                    "Id": "RollingUpdateType",
+                    "Name": "Rolling Update Type",
+                    "Description": "This includes three types: time-based rolling updates, health-based rolling updates, and immutable updates. Time-based rolling updates apply a PauseTime between batches. Health-based rolling updates wait for new instances to pass health checks before moving on to the next batch. Immutable updates launch a full set of instances in a new Auto Scaling group.",
+                    "Type": "String",
+                    "DefaultValue": "Time",
+                    "AllowedValues": [
+                        "Time",
+                        "Health",
+                        "Immutable"
+                    ],
+                    "AdvancedSetting": true,
+                    "Updatable": true,
+                    "DependsOn": [
+                        {
+                            "Id": "ElasticBeanstalkRollingUpdates.RollingUpdatesEnabled",
+                            "Value": true
+                        }
+                    ]
+                },
+                {
+                    "Id": "MaxBatchSize",
+                    "Name": "Max Batch Size",
+                    "Description": "The number of instances included in each batch of the rolling update.",
+                    "Type": "Int",
```

Comment:
Do we need some metadata to communicate with VS so it will allow null for numeric types?

Reply:
I think visual studio should have null handling from their side. I don't think its very beneficial to add additional data to our recipe to indicate that this could be null.